docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc

```diff
@@ -100,6 +100,18 @@ http_server_requests_seconds_sum{method="GET",outcome="SUCCESS",status="200",uri
 Use the new options `http-metrics-histograms-enabled` and `http-metrics-slos` to enable default histogram buckets or specific buckets for service level objectives (SLOs).
 Read more about histograms in the https://prometheus.io/docs/concepts/metric_types/#histogram[Prometheus documentation about histograms] on how to use the additional metrics series provided in `http_server_requests_seconds_bucket`.
 
+= Expected improvement in overall CPU usage and temporary increased database activity
+
+In {project_name} 24 release, we had a change in the password hashing algorithm which resulted in a increased CPU usage. To address that we opted to a different hashing algorithm Argon2 which brings the CPU usage back to where it was prior to the {project_name} 24 release.
+
+The Concepts for sizing CPU and memory resources in the {project_name} High Availability guide have been updated to reflect the new hashing defaults.
+
+After the upgrade, during a password-based login, the user's passwords will be re-hashed with the new hash algorithm and hash iterations as a one-off activity and updated in the database.
+As this clears the user from {project_name}'s internal cache, you'll also see an increased read activity on the database level.
+This increased database activity will decrease over time as more and more user's passwords have been re-hashed.
+
+To support the memory intensive nature of Argon2, we have updated the default GC from ParallelGC to G1GC for a better heap utilization.Please monitor the JVM heap utilization closely after this upgrade. Additional tuning may be necessary depending on your specific workload.
```

Comment:
I wonder if the GC change should have a separate heading to be more explicit. Also we might want to add some hints to the users to what behaviour should they expect with G1GC compared to ParallelGC?

Reply:
I'm currently reviewing these changes. I'll add a new sub-heading for this. For containerized environments, the overall behavior is similar compared to KC24. For non-containerized environments, we might need to run some more tests.

Reply:
We were debating over it, as the workload can vary based on the overrides each deployment might have and might not reflect the actual outcome, so we wanted it little open but hint there would be a change in behavior. But if you think it would help, we can specify what could be the difference when only defaults are used.